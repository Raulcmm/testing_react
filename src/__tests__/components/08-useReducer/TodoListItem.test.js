import { shallow } from "enzyme";
import { TodoListItem } from "../../../components/08-useReducer/TodoListItem";
import { demoTodos } from "../../fixtures/demoTodos";

describe("Pruebas a Todo List Item", () => {
	const handleDelete = jest.fn();
	const handleToggle = jest.fn();
	const index = 1;

	const wrapper = shallow(
		<TodoListItem
			todo={demoTodos[0]}
			index={index}
			handleDelete={handleDelete}
			handleToggle={handleToggle}
		/>
	);

	// // beforeEach(() => {
	// // 	//para resetear el llamado de funciones
	// // 	// jest.clearAllMocks();
	// // });
	test("debe mostrarse correctamente", () => {
		expect(wrapper).toMatchSnapshot();
	});

	test("Debe de llamar a la función handleDelete", () => {
		wrapper.find(".btn-danger").simulate("click");
		// expect(handleDelete).toHaveBeenCalledWith(expect.any(Function));
		expect(handleDelete).toHaveBeenCalledTimes(1); //y
	});

	test("Debe de llamar a la función handleToggle", () => {
		wrapper.find("p").simulate("click");
		expect(handleToggle).toHaveBeenCalledTimes(1); //y
	});

	test("debe mostar el texto correctamente", () => {
		expect(wrapper.find("p").text().trim()).toBe(
			`${index + 1}. ${demoTodos[0].desc}`
		);
	});

	test("Debe agregar la clase complete  si el done es true", () => {
		const todo = demoTodos[0];
		todo.done = true;
		const wrapper = shallow(
			<TodoListItem
				todo={todo}
				index={1}
				handleDelete={handleDelete}
				handleToggle={handleToggle}
			/>
		);

		expect(wrapper.find("p").hasClass("complete")).toBeTruthy();
	});
});
