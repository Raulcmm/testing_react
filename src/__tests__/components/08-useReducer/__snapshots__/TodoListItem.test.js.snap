// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a Todo List Item debe mostrarse correctamente 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="false"
    onClick={[Function]}
  >
    2
    . 
    Aprender react
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
